src/Sylius/Bundle/CoreBundle/Fixture/OrderFixture.php

```diff
@@ -216,34 +217,53 @@ private function address(OrderInterface $order, string $countryCode): void
 
     private function selectShipping(OrderInterface $order): void
     {
-        $shippingMethod = $this
-            ->faker
-            ->randomElement($this->shippingMethodRepository->findEnabledForChannel($order->getChannel()))
-        ;
-
-        foreach ($order->getShipments() as $shipment) {
-            $shipment->setMethod($shippingMethod);
-        }
-
         if ($this->orderShippingMethodSelectionRequirementChecker->isShippingMethodSelectionRequired($order)) {
+            $shippingMethod = $this
+                ->faker
+                ->randomElement($this->shippingMethodRepository->findEnabledForChannel($order->getChannel()))
+            ;
+
+            /** @var ChannelInterface $channel */
+            $channel = $order->getChannel();
+            Assert::notNull($shippingMethod, sprintf(
+                "No enabled shipping method was found for channel '%s'. " .
+                "Set 'skipping_shipping_step_allowed' option to true for this channel if you want to skip shipping method selection.",
+                $channel->getCode()
+            ));
+
+            foreach ($order->getShipments() as $shipment) {
+                $shipment->setMethod($shippingMethod);
+            }
+
             $this->applyCheckoutStateTransition($order, OrderCheckoutTransitions::TRANSITION_SELECT_SHIPPING);
+        } else {
```

Comment:
We could just use early return and skip `else` statement. Also, maybe reverse the `if` condition and put `$this->applyCheckoutStateTransition($order, OrderCheckoutTransitions::TRANSITION_SKIP_SHIPPING);` inside, which would make it easier to understand (IMHO). 

WDYT?